dbms/src/Storages/DeltaMerge/SegmentReadTask.cpp

```diff
@@ -893,4 +893,27 @@ void SegmentReadTask::updateMVCCIndexSize(ReadMode read_mode, size_t initial_ind
         !cache_key->is_version_chain && read_snapshot->delta->getSharedDeltaIndex()->getBytes() != initial_index_bytes)
         cache->setDeltaIndex(*cache_key, read_snapshot->delta->getSharedDeltaIndex());
 }
+
+UInt64 SegmentReadTask::estimatedBytesOfInternalColumns(const RSOperatorPtr & rs_operator, UInt64 start_ts) const
+{
+    auto real_ranges = segment->shrinkRowKeyRanges(ranges);
+    // stable->getDMFiles() at least return one DMFile.
+    auto handle_size = read_snapshot->stable->getDMFiles().front()->getColumnStat(MutSup::extra_handle_id).avg_size;
+    if (handle_size == 0)
```

Comment:
when will `handle_size` be zero?

Reply:
If stable is empty, `avg_size` could be zero.